php-frontend/src/test/java/org/sonar/php/PHPAnalyzerTest.java

```diff
@@ -150,11 +159,40 @@ public void log_error_and_continue_when_exception_in_terminate() throws Exceptio
     assertThat(logTester.logs(LoggerLevel.WARN)).contains("An error occurred while trying to terminate checks:");
   }
 
+  @Test
+  void testSuppressWarningFilterForMainAndTestFile() throws IOException {
+    PHPCheck check = new DummyCheck();
+    PHPCheck testCheck = new DummyCheck();
+    SuppressWarningFilter suppressWarningFilter = new SuppressWarningFilter();
+    PHPAnalyzer analyzer = createAnalyzer(List.of(check), List.of(testCheck), suppressWarningFilter);
+    File tmpFile = tmpFolder.newFile();
+    String fileUri = tmpFile.getCanonicalFile().toPath().toUri().toString();
+    InputFile file = FileTestUtils.getInputFile(tmpFile, "<?php\n//@SuppressWarnings(\"S11\")\n$a = 1;");
+
+    analyzer.nextFile(file);
+
+    List<PhpIssue> issues = analyzer.analyze();
+    assertThat(issues).hasSize(1);
+    assertThat(suppressWarningFilter.accept(fileUri, "S11", 3)).isFalse();
+    assertThat(suppressWarningFilter.accept(fileUri, "S11", 2)).isTrue();
+
+    suppressWarningFilter.reset();
+    issues = analyzer.analyzeTest();
+    assertThat(issues).hasSize(1);
+    assertThat(suppressWarningFilter.accept(fileUri, "S11", 3)).isFalse();
+    assertThat(suppressWarningFilter.accept(fileUri, "S11", 2)).isTrue();
+  }
+
   private PHPAnalyzer createAnalyzer(PHPCheck... checks) throws IOException {
     return createAnalyzer(Arrays.asList(checks),  Collections.emptyList());
   }
 
   private PHPAnalyzer createAnalyzer(List<PHPCheck> checks, List<PHPCheck> testFileChecks) throws IOException {
-    return new PHPAnalyzer(checks,  testFileChecks, tmpFolder.newFolder(), new ProjectSymbolData(), new DurationStatistics(sensorContext.config()), null);
+    return createAnalyzer(checks, testFileChecks, new SuppressWarningFilter());
+  }
+
+  private PHPAnalyzer createAnalyzer(List<PHPCheck> checks, List<PHPCheck> testFileChecks, SuppressWarningFilter suppressWarningFilter) throws IOException {
+    return new PHPAnalyzer(checks, testFileChecks, tmpFolder.newFolder(), new ProjectSymbolData(),
+      new DurationStatistics(sensorContext.config()), null, suppressWarningFilter);
```

Comment:
Nice improvement. But there is even more ;) 
I see that you use the second `createAnalyzer` method only in line 158 and 167. In both cases they use `new SuppressWarningFilter()`, so this can be further reduced to one method.